### PR TITLE
Document Packet Broker configuration

### PR DESCRIPTION
**Pull request by johanstokking**
#### Summary

Explain Packet Broker in more detail and steps to configure Packet Broker registration and manage routing policies.

Implementation is https://github.com/TheThingsNetwork/lorawan-stack/pull/3982

#### Changes

- Rewrite `/reference/packet-broker` section, preserving existing links

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links. Test with `HUGO_PARAMS_SEARCH_ENABLED=true` if style changes will affect the search bar.
- [x] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

File: doc/content/reference/configuration/packet-broker-agent.md [hunk 1]
```diff
@@ -3,37 +3,42 @@ title: "Packet Broker Agent Options"
...
+- `pba.forwarder.include-gateway-eui`: Indicates whether to include the gateway EUI in forwarded uplink messages {{< new-in-version "3.12.0" >}}
+- `pba.forwarder.include-gateway-id`: Indicates whether to include the gateway ID (plain or hashed) in forwarded uplink messages {{< new-in-version "3.12.0" >}}
+- `pba.forwarder.hash-gateway-id`: Indicates whether the forwarded gateway ID should be hashed. This setting is ineffective if `include-gateway-id` is `false` {{< new-in-version "3.12.0" >}}
 
 ## Home Network Options
 
 - `pba.home-network.enable`: Enable Home Network role
 - `pba.home-network.dev-addr-prefixes`: DevAddr prefixes to subscribe to
 - `pba.home-network.worker-pool.limit`: Limit of active workers concurrently processing uplink messages and publishing downlink messages
-- `pba.home-network.blacklist-forwarder`: Blacklist traffic from Forwarder to avoid traffic loops. Enable this when you have the Forwarder role enabled and overlapping forwarding ranges in `gs.forward` (see [configuration]({{< relref "gateway-server.md" >}})). Only disable this for testing.
```
Comment (johanstokking) [inline doc/content/reference/configuration/packet-broker-agent.md:39]: This was already deprecated and is now removed

File: doc/content/reference/configuration/packet-broker-agent.md [hunk 2]
```diff
@@ -3,37 +3,42 @@ title: "Packet Broker Agent Options"
...
+- `pba.iam-address`: Address of Packet Broker IAM {{< new-in-version "3.12.0" >}}
+- `pba.control-plane-address`: Address of Packet Broker Control Plane {{< new-in-version "3.12.0" >}}
+- `pba.data-plane-address`: Address of Packet Broker Data Plane. See [Packet Broker Clients](https://github.com/packetbroker/pb) for available cluster addresses
+- `pba.insecure`: Connect without using TLS (only for test environments)
 - `pba.net-id`: LoRa Alliance NetID
 - `pba.tenant-id`: Tenant ID within the NetID
 - `pba.cluster-id`: Cluster ID uniquely identifying this cluster within a NetID and tenant. The cluster ID is used for shared subscriptions (i.e. splitting traffic over multiple Packet Broker Agents) and as Forwarder ID to route downlink traffic to the right cluster
+- `pba.home-network-cluster-id`: Home Network Cluster ID, if different from the Cluster ID. Leave empty to fallback to `cluster-id`
 
-## Client TLS Options
```
Comment (johanstokking) [inline doc/content/reference/configuration/packet-broker-agent.md:13]: mTLS is not supported by Packet Broker anymore, so we don't have to mark this deprecated or anything

File: doc/content/reference/packet-broker/connect.md [hunk 3]
```diff
@@ -1,22 +1,21 @@
...
+{{% tts %}} Cloud and The Things Network are already connected to Packet Broker. Use this section to connect your {{% tts %}} Enterprise or Open Source deployment to Packet Broker.
 
-## Peering Between {{% tts %}} Cloud and The Things Network
-
-By default, Packet Broker enables peering in the following scenario:
-
-- Traffic received on The Things Network is forwarded to The Things Stack Cloud
-
-Currently, Packet Broker does not peer by default between {{% tts %}} Cloud tenants, or forward traffic from {{% tts %}} Cloud to The Things Network. Cloud customers may file a support ticket to enable this feature, and it will be possible for users to configure {{% tts %}} to support peering from {{% tts %}} Cloud in an upcoming release.
+If you are using {{% tts %}} Cloud or if your {{% tts %}} deployment is connected to Packet Broker, [proceed to configuration]({{< relref "configure" >}}).
```
Comment (nejraselimovic) [inline doc/content/reference/packet-broker/connect.md:17]:
```suggestion
If you are using {{% tts %}} Cloud or if your {{% tts %}} deployment is connected to Packet Broker, proceed to [configure the Packet Broker]({{< relref "configure" >}}).
```

**Review by nejraselimovic — APPROVED** [review]

**Comment by johanstokking** [comment]
@nejraselimovic thanks!

**Pull request merged**